homeworks/week6/Tam-Tran.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""
+// Given a string S and a string T, find the minimum window in S which will contain all the characters in T
+// Input: S = "xyyzyzyx" T = "xyz”		Output: "zyx"
+"""
+
+from collections import Counter
+
+def check_window_valid(current_count, expected_count):
```

Comment:
The complexity of this method is O(M) where M = T.length
So the algorithm complexity should be O(M*N).

The optimal solution can be O(M+N).